### PR TITLE
fix(electrostatics): restore reciprocal strain virial

### LINKED ISSUE #136: 🐛[BUG]: `ewald_reciprocal_space` autograd silently returns a wrong virial
### Version

0.4.0

### On which installation method(s) does this occur?

_No response_

### Describe the issue

Taking a strain/cell gradient through `ewald_reciprocal_space` produces a wrong virial, with no error and no warning. `compute_virial=True` returns the correct tensor, but it is deprecated.

### Minimum reproducible example

```python
import warnings

import torch
from nvalchemiops.torch.interactions.electrostatics import (
    ewald_reciprocal_space,
    generate_k_vectors_ewald_summation,
)

torch.set_default_dtype(torch.float64)

# 4 charges in a triclinic box
CELL = torch.tensor([[6.0, 0.0, 0.0], [1.0, 5.0, 0.0], [0.5, 0.7, 5.5]])
POS = torch.tensor([[0.5, 0.5, 0.5], [3.0, 1.0, 2.0], [1.5, 3.5, 4.0], [4.5, 2.5, 1.0]])
CHARGES = torch.tensor([1.0, -1.0, 0.7, -0.7])
ALPHA = torch.tensor([0.4])
K_CUTOFF = torch.tensor([4.0])

# built once from the unstrained cell: a constant w.r.t. the strain
K_PRECOMPUTED = generate_k_vectors_ewald_summation(cell=CELL, k_cutoff=K_CUTOFF)


def energy(pos, cell, k_vectors=None):
    """`k_vectors=None` rebuilds them from the strained `cell` (requires_grad)."""
    if k_vectors is None:
        k_vectors = generate_k_vectors_ewald_summation(cell=cell, k_cutoff=K_CUTOFF)
    return ewald_reciprocal_space(positions=pos, charges=CHARGES, cell=cell,
                                  k_vectors=k_vectors, alpha=ALPHA).sum()


def strain(eps):
    """Affine deformation of positions and cell by a symmetric strain."""
    sym = 0.5 * (eps + eps.T)
    return POS + POS @ sym, CELL + CELL @ sym


def virial_finite_difference(h=1e-6):
    """Reference, with the k-vectors following the strained cell."""
    dE = torch.zeros(3, 3)
    for a in range(3):
        for b in range(3):
            values = []
            for s in (+h, -h):
                eps = torch.zeros(3, 3)
                eps[a, b] += 0.5 * s
                eps[b, a] += 0.5 * s
                with torch.no_grad():
                    values.append(energy(*strain(eps)))
            dE[a, b] = (values[0] - values[1]) / (2 * h)
    return -dE  # nvalchemiops convention: W = -dE/d(strain)


def virial_autograd(k_vectors=None):
    eps = torch.zeros(3, 3, requires_grad=True)
    return -torch.autograd.grad(energy(*strain(eps), k_vectors), eps)[0]


def virial_kernel():
    with warnings.catch_warnings():  # the flag is deprecated
        warnings.simplefilter("ignore", DeprecationWarning)
        out = ewald_reciprocal_space(
            positions=POS, charges=CHARGES, cell=CELL.unsqueeze(0),
            k_vectors=K_PRECOMPUTED, alpha=ALPHA,
            batch_idx=torch.zeros(POS.shape[0], dtype=torch.int32),
            compute_virial=True,
        )
    return out[-1].reshape(3, 3)


def show(label, W, reference=None):
    verdict = ""
    if reference is not None:
        error = (W - reference).abs().max()
        verdict = f"max error = {error:.1e}  {'ok' if error < 1e-8 else 'WRONG'}"
    print(f"\n{label:<44}{verdict}")
    for row in W:
        print("   " + "  ".join(f"{v:+12.6f}" for v in row))


reference = virial_finite_difference()

show("finite difference (reference)", reference)
show("autograd, pre-computed k-vectors", virial_autograd(K_PRECOMPUTED), reference)
show("autograd, k-vectors from strained cell", virial_autograd(), reference)
show("compute_virial=True (deprecated)", virial_kernel(), reference)
```

### Relevant log output

```shell
finite difference (reference)               
      -0.419709     +0.108878     +0.022028
      +0.108878     +0.062030     -0.020228
      +0.022028     -0.020228     -0.061639

autograd, pre-computed k-vectors            max error = 5.8e-01  WRONG
      +0.165091     +0.026475     -0.049095
      +0.026475     +0.118494     -0.053209
      -0.049095     -0.053209     +0.087324

autograd, k-vectors from strained cell      max error = 5.8e-01  WRONG
      +0.165091     +0.026475     -0.049095
      +0.026475     +0.118494     -0.053209
      -0.049095     -0.053209     +0.087324

compute_virial=True (deprecated)            max error = 7.6e-11  ok
      -0.419709     +0.108878     +0.022028
      +0.108878     +0.062030     -0.020228
      +0.022028     -0.020228     -0.061639
```

### Environment details

```shell

```

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit-Ops Pull Request

## Description

Fix Torch reciprocal Ewald cell and strain autograd by preserving graph-connected reciprocal vectors when both the cell and vectors require gradients. Emit an eager advisory warning when fixed Cartesian vectors are paired with cell differentiation because that derivative is not the physical Ewald strain virial.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

Fixes #136

## Changes Made

- Preserve the supplied reciprocal-vector autograd graph in `ewald_reciprocal_space` when cell gradients are requested.
- Warn in eager execution when fixed Cartesian reciprocal vectors are used with a differentiable cell.
- Add CPU, CUDA, finite-difference, direct-virial, compile, geometry-dependent charge, and double-backward regression coverage.
- Document differentiable-cell reciprocal-vector usage and update the changelog.

## Testing

- [x] Unit tests pass locally (`make pytest`) — full suite not run
- [x] Linting passes (`make lint`) — not run
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes — N/A, no new functions or classes
- [x] I have updated the documentation (if applicable)

## Additional Notes

None.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

**Comment by laserkelvin** [comment]
/ok to test [aa2a02f](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/138/commits/aa2a02f2107f803635587d60f0e42bf7065d3e92)

File: docs/userguide/about/migration.md [hunk 1]
```diff
@@ -52,10 +52,16 @@ entry points instead of regenerating it inside hot loops.
 
 These inputs are caches, not differentiable parameters. `alpha`, cutoffs,
 mesh controls, batch metadata, neighbor topology, and PME B-spline moduli are
-treated as constants even if supplied as grad-bearing tensors. Cell-derived
-caches such as `k_vectors`, `k_squared`, `volume`, and `cell_inv_t` remain
-accepted when cell derivatives are requested, but they are static metadata
-assumed to correspond to the current `cell`.
+treated as constants even if supplied as grad-bearing tensors. For
+`ewald_summation`, caller-supplied `k_vectors` remain static metadata assumed
```
Comment (laserkelvin) [inline docs/userguide/about/migration.md:104]: suggestion: `ewald_summation` intentionally keeps supplied `k_vectors` static, but its `k_vectors` parameter docstring still only says “Pre-computed reciprocal lattice vectors.” I assume callers should be able to discover this gradient contract from the API reference, not only the migration guide. Please state there that supplied vectors are static and recommend omitting `k_vectors` for physical cell derivatives.

File: test/interactions/electrostatics/bindings/torch/test_ewald.py [hunk 2]
```diff
@@ -8201,6 +8331,11 @@ def energy_fn(p, q, c):
         elif part == "recip":
 
             def energy_fn(p, q, c):
+                k_vectors = generate_k_vectors_ewald_summation(
```
Comment (laserkelvin) [inline test/interactions/electrostatics/bindings/torch/test_ewald.py:9792]: suggestion: Regenerating `k_vectors` here moves the reciprocal second-order tests to the connected-k path, so the still-supported fixed-k cell/stress path loses its previous second-order coverage. I assume both derivative modes remain supported. Please retain this case and add one focused fixed-k stress-loss or cell `gradgrad` regression.

File: nvalchemiops/torch/interactions/electrostatics/ewald.py [hunk 3]
```diff
@@ -1058,6 +1065,26 @@ def ewald_reciprocal_space(
     losses and higher-order derivatives recompute safe partials or connected
     gradients as needed to avoid double-counting that chain term (issue #115).
     """
+    allow_cell_grad_with_k_vectors = bool(
```
Comment (laserkelvin) [inline nvalchemiops/torch/interactions/electrostatics/ewald.py:1317]: nit: This establishes public k-vector gradients, but the internal comment at lines 574–577 still says public leaf gradients are outside the contract and the cell path is valid only for vectors generated inside the full wrapper. I assume comments should match this new component policy. Please update that comment with the contract change.

File: docs/userguide/about/migration.md [hunk 4]
```diff
@@ -52,10 +52,16 @@ entry points instead of regenerating it inside hot loops.
...
+to correspond to the current `cell`.
+
+The lower-level Torch `ewald_reciprocal_space` component preserves a supplied
```
Comment (laserkelvin) [inline docs/userguide/about/migration.md:107]: suggestion: This component exception conflicts with the Torch module overview, which still says all cell-derived `k_vectors` are static and their generation derivatives are never recovered. I assume the module page should describe the same public contract. Please scope `docs/modules/torch/electrostatics.rst:11–16` to the full Ewald/PME APIs and call out this component behavior.

**Review by laserkelvin — COMMENTED** [review]
Four focused suggestions on the API documentation and preserving fixed-k second-order coverage; details inline.

**Review by laserkelvin — APPROVED** [review]
Generally fine - just minor things to address

File: docs/userguide/about/migration.md (hunk 1 above)
Comment (zubatyuk) [reply docs/userguide/about/migration.md:104]: Fixed in 464bfbc0a01b91037de875bebf992d701f361a36.

File: nvalchemiops/torch/interactions/electrostatics/ewald.py (hunk 3 above)
Comment (zubatyuk) [reply nvalchemiops/torch/interactions/electrostatics/ewald.py:1317]: Fixed in 464bfbc0a01b91037de875bebf992d701f361a36.

File: docs/userguide/about/migration.md (hunk 4 above)
Comment (zubatyuk) [reply docs/userguide/about/migration.md:107]: Fixed in 464bfbc0a01b91037de875bebf992d701f361a36.

File: test/interactions/electrostatics/bindings/torch/test_ewald.py (hunk 2 above)
Comment (zubatyuk) [reply test/interactions/electrostatics/bindings/torch/test_ewald.py:9792]: Fixed in 464bfbc0a01b91037de875bebf992d701f361a36.

**Comment by laserkelvin** [comment]
/ok to test [7b27f1d](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/138/commits/7b27f1d818a4ece887896921ed22745464e6f368)

**Review by dallasfoster — APPROVED** [review]
